### PR TITLE
update node tests

### PR DESCRIPTION
**Pull request by simon-id**

### CODE REVIEW

File: scenarios/appsec/test_customconf.py [hunk 1]
```diff
@@ -37,7 +37,7 @@ def test_c05(self):
         stdout.assert_presence(r"AppSec could not read the rule file .* as it was invalid: .*", level="CRITICAL")
 
 
-@released(java="0.93.0", nodejs="2.4.1", php_appsec="0.3.0", ruby="?")
+@released(java="0.93.0", php_appsec="0.3.0", ruby="?")
```
Comment (cbeauchesne) [inline scenarios/appsec/test_customconf.py:40]: I do not understand, you're saying here that this feature is working for all nodejs tracer version Oo

File: tests/appsec/waf/test_rules.py [hunk 2]
```diff
@@ -293,14 +293,18 @@ def test_nosqli_value(self):
...
     @irrelevant(context.appsec_rules_version < "1.3.0", reason="before 1.3.0, keys was not supported")
-    def test_nosqli_keys(self):
-        """AppSec catches NoSQLI attacks"""
+    @irrelevant(library="nodejs", reason="Node interprets brackets as arrays, so they're truncated")
+    def test_nosqli_keys_with_brackets(self):
+        """AppSec catches NoSQLI attacks with brackets"""
         r = self.weblog_get("/waf/", params={"[$ne]": "value"})
         interfaces.library.assert_waf_attack(r, rules.nosql_injection.crs_942_290)
 
+    @irrelevant(context.libddwaf_version < "1.1.0", reason="Key matching is not supported before this version")
```
Comment (cbeauchesne) [inline tests/appsec/waf/test_rules.py:304]: Unfortunately, `libddwaf_version` is not extracted for all tracers, you need to skip it for  each tracer individually

**Review by cbeauchesne — CHANGES_REQUESTED** [review]

File: scenarios/appsec/test_customconf.py (hunk 1 above)
Comment (cbeauchesne) [reply scenarios/appsec/test_customconf.py:40]: a `?`

File: tests/appsec/waf/test_rules.py [hunk 3]
```diff
@@ -295,14 +295,19 @@ def test_nosqli_value(self):
...
+    @missing_feature(context.library in ["golang", "php", "ruby"], reason="Need to use last WAF version")
     @missing_feature(context.library < "java@0.96.0", reason="Was using a too old WAF version")
     @irrelevant(context.appsec_rules_version < "1.3.0", reason="before 1.3.0, keys was not supported")
     def test_nosqli_keys(self):
         """AppSec catches NoSQLI attacks in keys"""
         r = self.weblog_get("/waf/", params={"[$ne]": "value"})
         interfaces.library.assert_waf_attack(r, rules.nosql_injection.crs_942_290)
 
+    @irrelevant(context.libddwaf_version < "1.1.0", reason="Key matching is not supported before this version")
```
Comment (cbeauchesne) [inline tests/appsec/waf/test_rules.py:307]: Just remove this line

File: scenarios/appsec/test_customconf.py [hunk 4]
```diff
@@ -38,7 +38,7 @@ def test_c05(self):
         stdout.assert_presence(r"AppSec could not read the rule file .* as it was invalid: .*", level="CRITICAL")
 
 
-@released(java="0.93.0", nodejs="2.4.1", php_appsec="0.3.0", ruby="?")
+@released(java="0.93.0", nodejs="?", php_appsec="0.3.0", ruby="?")
```
Comment (cbeauchesne) [inline scenarios/appsec/test_customconf.py:41]: You deactivated the test, it's on purpose ?

Comment (simon-id) [reply scenarios/appsec/test_customconf.py:41]: Yes, it's not implemented

**Review by cbeauchesne — APPROVED** [review]

**Pull request merged**